sdk/communication/CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+
+The Azure Communication Services SDK for JS currently consists of six different packages. While each package has its own set of environment variables to make their tests run successfully, all of them follow a similar structure that allows a smooth onboarding process.
+
+Let's get started with how to set up the repo itself.
+
+## Building
+
+Install and link all dependencies `rush update`, and build the code base `rush rebuild`.
+You can then `rush build -t <packagename>` to build a single project.
+Once the package has been built, let's jump on how to run the tests to see that everything is in order.
+
+## Testing
+
+Make sure to check out the general contributing guide the Azure SDK repo has for a more in-depth look at testing and setting up your dev environment. You can check out the contributing file [here](https://github.com/Azure/azure-sdk-for-js/blob/master/CONTRIBUTING.md)
+
+When you go inside the package you are working with, you will see a folders called `recordings\browsers` and `recordings\node`. In JS we performs tests in the node and browser environment. Each folder contains, as its name suggests, recordings of successful calls to the API that allow us to run the tests in PLAYBACK mode and remove the necessity of hitting the actual resources every time we may want to test.
+
+To test with environment variables, JS has a `sample.env` for developers to use as a template. You can rename the file to `.env` and add the appropriate environment variables.
+
+### Playback mode
+
+By default tests are run in playback mode. You can also set an environment variable in your `.env` called `TEST_MODE` and set its value to `PLAYBACK`. You can then run `rushx test` command to run the tests.
+
+If the tests are successful, we can proceed to run the tests in LIVE mode.
+
+### Live mode
+
+Since in LIVE mode we are hitting an actual resource, we must set the appropriate environment variables to make sure the code tests against the resource we want. Set up an env variable called `AZURE_COMMUNICATION_LIVETEST_CONNECTION_STRING` for Phone Number and SMS packages and set it to the connection string of the resource you want to test against. For all other communication packages, set up an environment variable called `COMMUNICATION_CONNECTION_STRING` and set it to the connection string of the resource you want to test against.
+
+Depending on which package you are testing, it may need special environment variables to test succesfully. In each package, there is a `recordedClient.ts` file; In that file you will find `const replaceableVariables` with the names of the variables the package uses. Make sure to set these variables before running the tests themselves.
```

Comment:
Ideally these would be defined in the `test.env`. This way when they create a `.env`  from the template it will already include the necessary vars for tests to run. Just looking at the phone numbers package, the `test.env` is missing some required env vars. If we make that change this step will become unecessary.

Reply:
I was debating between having them look at the `test.env` or actually looking into the `recordedClient.ts`. I found that sometimes the `test.env` wasn't always kept up to date so worried there would be a bit of misdirection leading them there. Thats why I opted to show them where the variables were actually being replaced cause this feels like it would never go wrong / out of sync.